pkg/audit/verifier.go

```diff
@@ -38,7 +38,7 @@ type Verifier struct {
 }
 
 type downloader interface {
-	DownloadShares(ctx context.Context, pointer *pb.Pointer, stripeIndex int, authorization *pb.SignedMessage) (shares []share, nodes []*pb.Node, err error)
+	DownloadShares(ctx context.Context, pointer *pb.Pointer, stripeIndex int, authorization *pb.SignedMessage) (shares map[int]share, nodes map[int]*pb.Node, err error)
```

Comment:
I would prefer that these are kept as slices for probably a little better memory usage, but if this makes fixing the bug easier, it's fine for now.

Reply:
It's not about easier, but about more efficient. If we keep them as slices then, for example [here](https://github.com/storj/storj/pull/713/files#diff-d539f8f3a5159250dae627f3422b0a16R237), we need to iterate through the `nodes` slice to find the node with the respective piece number.